controllers/operator_controller.go

```diff
@@ -71,15 +71,15 @@ func (r *OperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	compareOp := reconciledOp.DeepCopy()
 	existingOp.Status, compareOp.Status = operatorsv1alpha1.OperatorStatus{}, operatorsv1alpha1.OperatorStatus{}
 	existingOp.Finalizers, compareOp.Finalizers = []string{}, []string{}
-	specDiffers := !equality.Semantic.DeepEqual(existingOp, compareOp)
+	unexpectedFieldsChanged := !equality.Semantic.DeepEqual(existingOp, compareOp)
```

Comment:
could we create a function that takes the `reconciledOp` and returns if an unexpected field was changed? it may make things easier and would allow us to update the block above to look like:
```
updateStatus := !equality.Semantic.DeepEqual(existingOp.Status, reconciledOp.Status)
updateFinalizers := !equality.Semantic.DeepEqual(existingOp.Finalizers, reconciledOp.Finalizers)
unexpectedFieldsChanged := checkForUnexpectedFieldChanges(reconciledOp) # Function name could be different.
```